### PR TITLE
fix(relay): remove unconditional `async-std` dependency

### PR DESCRIPTION
**Pull request by quininer**
## Description

`libp2p-relay` crate specifies the `libp2p-swarm/async-std` feature, which causes an `async-std` dependency to always be introduced.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

**Review by thomaseizinger — APPROVED** [review]
Thank you! Nice catch.

Can you patch-bump `libp2p-relay` please? Otherwise we can't release it. Plus a short changelog entry would be good.

You will also have to commit the log-file change and bump the relay version in the main mainfest (CI will tell you if you mess it up :smile:  )

**Comment by quininer** [comment]
@thomaseizinger I updated CHANGELOG, and I noticed that 0.16.1 has not been released yet, so I didn't update the patch version.

File: protocols/relay/CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,19 @@
 ## 0.16.1 - unreleased
 
-- Export `RateLimiter` type. 
+- Export `RateLimiter` type.
   See [PR 3742].
 
 - Add functions to access data within `Limit`.
   See [PR 4162].
 
+- remove unconditional `async-std` dependency.
```
Comment (thomaseizinger) [inline protocols/relay/CHANGELOG.md]:
```suggestion
- Remove unconditional `async-std` dependency.
```

**Review by thomaseizinger — APPROVED** [review]
Thanks!

**Review by mxinden — COMMENTED** [review]
:pray: thank you for the fix!

**Pull request merged**